cmd/tempo-cli/cmd-gen-bloom-index.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/google/uuid"
+
+	"github.com/grafana/tempo/tempodb/backend"
+	"github.com/grafana/tempo/tempodb/encoding"
+	"github.com/grafana/tempo/tempodb/encoding/common"
+)
+
+type genIndexCmd struct {
+	TenantID string `arg:"" help:"tenant-id within the bucket"`
+	BlockID  string `arg:"" help:"block ID to list"`
+	backendOptions
+}
+
+func ReplayBlockAndGetRecords(meta *backend.BlockMeta, filepath string) ([]common.Record, error, error) {
+	v, err := encoding.FromVersion(meta.Version)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var warning error
+	// replay file to extract records
+	f, err := os.OpenFile(filepath, os.O_RDONLY, 0644)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	dataReader, err := v.NewDataReader(backend.NewContextReaderWithAllReader(f), meta.Encoding)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer dataReader.Close()
+
+	var buffer []byte
+	var records []common.Record
+	objectReader := v.NewObjectReaderWriter()
+	currentOffset := uint64(0)
+	for {
+		buffer, pageLen, err := dataReader.NextPage(buffer)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			warning = err
+			break
+		}
+
+		reader := bytes.NewReader(buffer)
+		id, _, err := objectReader.UnmarshalObjectFromReader(reader)
+		if err != nil {
+			warning = err
+			break
+		}
```

Comment:
i believe this needs to loop through all objects in the page? it seems to only get the first. basically loop until `UnmarshalObjectFromReader` returns io.EOF.

To test we should create a block using the StreamingBlock, replay it using this code and then compare the index/blooms.

Reply:
> To test we should create a block using the StreamingBlock, replay it using this code and then compare the index/blooms.

I don't want to do this because it will require that we download the block from the remote backend. It should be easy to write a verification utility that iterates through the index and ensures the records are lined up accordingly. WDYT?

Reply:
No, I mean to just create use a StreamingBlock to create a block, write it to a local backend and then use this code to recreate the index/bloom and make sure it's correct.

This test does that but then uses a BackendBlock to read the streaming block:
https://github.com/grafana/tempo/blob/main/tempodb/encoding/streaming_block_test.go#L150